src/test_driver/openiotsdk/integration-tests/all-clusters-app/test_app.py

```diff
@@ -0,0 +1,176 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import pytest
+from time import sleep
+
+from common.utils import *
+
+from chip.clusters.Objects import DoorLock
+
+import logging
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def binaryPath(request, rootDir):
+    if request.config.getoption('binaryPath'):
+        return request.config.getoption('binaryPath')
+    else:
+        return os.path.join(rootDir, 'examples/all-clusters-app/openiotsdk/build/chip-openiotsdk-all-clusters-app-example.elf')
+
+
+@pytest.mark.smoketest
+def test_smoke_test(device):
+    ret = device.wait_for_output("Open IoT SDK all-clusters-app example application start")
+    assert ret != None and len(ret) > 0
+    ret = device.wait_for_output("Open IoT SDK all-clusters-app example application run")
+    assert ret != None and len(ret) > 0
+
+
+@pytest.mark.commissioningtest
+def test_commissioning(device, controller):
+    assert controller != None
+    devCtrl = controller
+
+    setupPayload = get_setup_payload(device)
+    assert setupPayload != None
+
+    commissionable_device = discover_device(devCtrl, setupPayload)
+    assert commissionable_device != None
+
+    assert commissionable_device.vendorId == int(setupPayload.attributes['VendorID'])
+    assert commissionable_device.productId == int(setupPayload.attributes['ProductID'])
+    assert commissionable_device.addresses[0] != None
+
+    nodeId = connect_device(devCtrl, setupPayload, commissionable_device)
+    assert nodeId != None
+    log.info("Device {} connected".format(commissionable_device.addresses[0]))
+
+    ret = device.wait_for_output("Commissioning completed successfully")
+    assert ret != None and len(ret) > 0
+
+    assert disconnect_device(devCtrl, nodeId)
+
+
+LOCK_CTRL_TEST_PIN_CODE = 12345
+LOCK_CTRL_TEST_USER_INDEX = 1
+LOCK_CTRL_TEST_ENDPOINT_ID = 1
+LOCK_CTRL_TEST_USER_NAME = 'testUser'
+LOCK_CTRL_TEST_CREDENTIAL_INDEX = 1
+
+
+@pytest.mark.ctrltest
+def test_lock_ctrl(device, controller):
```

Comment:
I believe we should test for some other clusters.

Reply:
Agree, but I'm not sure which one. Not all clusters have implemented handles that allow us to perform actions useful for integration tests. I'm rather thinking about some basic clusters such as `Basic` or 'AccessControl' clusters that can be validated here. Feature-specific clusters will be validated in other examples. What do you think about it?

Reply:
The more I think about it, the less my remark makes sense as the certification test suite should validate other clusters. 
Let's keep this as a smoke test. Details will be validated by the certification test suite.